package-res/resources/web/pentaho/type/_doc/IConfigurationService.jsdoc

```diff
@@ -0,0 +1,137 @@
+ * a given type and context variables,
+ * sorts these by specificity,
+ * merges the rules' type specifications and, finally,
+ * returns the result.
+ * For more information on the specificity of rules,
+ * see [spec.ITypeConfiguration]{@link pentaho.type.spec.ITypeConfiguration}.
+ *
+ * The merged type configuration specification can then be used to configure a value type,
+ * by calling its [pentaho.type.Type.implement]{@link pentaho.lang.Base.implement} method.
+ *
+ * Usually, you'll not call these operations directly.
+ *
+ * The singleton configuration service, {@link pentaho.type.configurationService},
+ * already handles loading of all type configurations registered
+ * with the [pentaho/service]{@link pentaho.service} module,
+ * under the service id `pentaho.type.spec.ITypeConfiguration`.
+ *
+ * Then, the [Context]{@link pentaho.type.Context} class
+ * already handles obtaining configurations from its configuration service (by default, the singleton one),
+ * and applying these to the requested value types.
+ *
+ * The following example illustrates how an implementation of this service would be used:
+ *
+ * ```js
+ * var myTypeConfig = {
+ *   rules: [
+ *     // Disable a still experimental Viz.
+ *     {
+ *       select: {
+ *         type: "my/radial/bar"
+ *       },
+ *       apply: {
+ *         isBrowsable: false
+ *       }
+ *     },
+ *
+ *     // Enable it, only for the dev user, "john", when in Analyzer
+ *     {
+ *       select: {
+ *         type:      "my/radial/bar",
+ *         user:      "john",
+ *         container: "pentaho.analyzer"
+ *       },
+ *       apply: {
+ *         isBrowsable: true
+ *       }
+ *     }
+ *   ]
+ * };
+ *
+ * var configService = new SomeConfigurationService();
+ *
+ * configService.add(myTypeConfig);
+ *
+ * var mergedTypeConfigSpec = configService.select("pentaho/visual/ccc/bar", {
+ *      user:      "john",
+ *      container: "pentaho.cdf"
+ *    });
+ *
+ * // Results in a specification like:
+ * // {
+ * //   isBrowsable: false
+ * // }
+ * ```
+ *
+ * @name pentaho.type.IConfigurationService
+ * @interface
+ *
+ * @see pentaho.type.Context
+ * @see pentaho.type.spec.ITypeConfiguration
+ */
+
+/**
+ * Obtains the merged configuration specification of
+ * the selection of configuration rules that apply to a given type and context variables.
+ *
+ * @name select
+ * @memberOf pentaho.type.IConfigurationService#
+ * @method
+ *
+ * @param {string} typeId The [id]{@link pentaho.type.Type#id} of the type whose
+ * configuration specification is desired.
+ *
+ * @param {pentaho.type.IContextVars} [contextVars] The context variables that are used to select rules.
+ *
+ * A {@link Nully} variable value is equivalent to a variable that is not present.
+ * It matches only configuration rules that do not select that variable.
+ *
+ * When the map is unspecified, it is like every variable had been specified with a `null` value.
```

Comment:
Suggest:
When the map is unspecified, every variable will appear as though it had been specified with a 'null' value.